tempo_embeddings/embeddings/vector_database.py

```diff
@@ -48,15 +49,31 @@ def connect(self):
 
     @abstractmethod
     def insert_corpus(
-        self, collection: Collection, corpus: Corpus
+        self, collection: Union[Collection, str], corpus: Corpus
```

Comment:
I don't understand (yet) in which cases a string is passed here. Adding a docstring might clarify that (and would be helpful for future reference anyway).

Reply:
this is an addition to the abstract method because Chroma API requires a collection object, whereas Weaviate requires only the collection name (as a string) since each API call to Weaviate needs a context manager that opens and closes the connection immediately.